src/App.js

```diff
@@ -1,26 +1,62 @@
 import React from 'react';
-import logo from './logo.svg';
 import './App.css';
+import LevelNav from './LevelNav.js'
+import LevelMenu from './LevelMenu.js'
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+
+import { Route, BrowserRouter, Switch } from 'react-router-dom';
+
+
+class App extends React.Component{
+  state = { open: false }
+  toggleOpen = () => {
+    this.setState({
+      open: !this.state.open
+    })
+  }
+  render() {
+    return (
+        <BrowserRouter>
+          <div className="App">
+            <div className="Header">
+              <span className="Title"> Selector Safari </span>
+              <div className="NavButtons">
+                <a href="#/" className="TutorialButton"> Tutorial </a>
+                <Switch>
+                  <Route path="/level/1"
+                         render={(props) => (
+                          <LevelNav toggle={this.toggleOpen} show={this.state.open} levelNum="1"></LevelNav>
+                         )}
+                  />
+                  <Route path="/level/2"
+                         render={(props) => (
+                             <LevelNav toggle={this.toggleOpen} show={this.state.open} levelNum="2"></LevelNav>
+                         )}
+                  />
+                  <Route path="/level/3"
+                         render={(props) => (
+                             <LevelNav toggle={this.toggleOpen} show={this.state.open} levelNum="3"></LevelNav>
+                         )}
+                  />
+                  <Route path="/level/4"
+                         render={(props) => (
+                             <LevelNav toggle={this.toggleOpen} show={this.state.open} levelNum="4"></LevelNav>
+                         )}
+                  />
+                  <Route path="/"
+                         render={(props) => (
+                             <LevelNav toggle={this.toggleOpen} show={this.state.open} levelNum="1"></LevelNav>
+                             )}
+                  />
```

Comment:
In the beginning I said you wouldn't need state to keep track of which level we're on, but that was a mistake, sorry about that 🤦 Keeping the current level in state will help us reduce duplication here so that we only have to handle the path /level/:levelNum, instead of keeping a separate Route for each level. This also applies to LevelMenu.js.